pilot/pkg/features/pilot.go

```diff
@@ -333,6 +333,13 @@ var (
 	EnableEDSCaching = env.RegisterBoolVar("PILOT_ENABLE_EDS_CACHE", true,
 		"If true, Pilot will cache EDS responses.").Get()
 
+	EnableEDSCacheMetrics = env.RegisterBoolVar("PILOT_ENABLE_EDS_CACHE_STATS", true,
+		"If true, Pilot will collect metrics for EDS cache efficency.").Get()
+
+	EDSCacheMaxSize = env.RegisterIntVar("PILOT_ENABLE_EDS_CACHE_STATS", 2000,
```

Comment:
Env var is wrong, and same comment about EDS -> XDS.

Can we make this higher by default? Based on https://prow.istio.io/view/gs/istio-prow/logs/benchmark-report_istio_postsubmit/621, a large message size is about 5kb. 2k puts the memory cap on this around 10mb. We can allocate FAR more than this I think. The cache has massive impact on memory and CPU usage. If we don't cache, we generate the same resource 100x in parallel, and Nx per push (n = proxies), which ends up causing memory to spike during pushes, which in turn adds GC pressure.

I think 20k is sufficient starting place?

Reply:
This definitely needs performance testing of its own before release, maybe with an eviction counter but ty for the suggestion.

Reply:
Should we rename the `PILOT_ENABLE_EDS_CACHE` as well?

Reply:
Oh I forgot we had that. Yes lets rename it, we didn't ship it yet so no problem changing